url5a01f2c1d9_core9_core_tgz-pJ8-io_core9_core_boot_BootstrapFrameworkJ8/src/net/xeoh/plugins/base/util/OptionUtils.java

```diff
@@ -0,0 +1,149 @@
+ * documentation and/or other materials provided with the distribution.
+ * 
+ * Neither the name of the author nor the names of its contributors may be used to endorse or
+ * promote products derived from this software without specific prior written permission.
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS
+ * OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR
+ * TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+ * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ * 
+ */
+package net.xeoh.plugins.base.util;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import net.xeoh.plugins.base.Option;
+
+/**
+ * Handles options within plugin methods. Likely to be replaced by 
+ * <a href="http://jcores.net">jCores</a>.
+ * 
+ * @author Ralf Biedert
+ * 
+ * @param <T> Type paramter.
+ */
+public class OptionUtils<T extends Option> {
+
+    /** Options */
+    private T[] options;
+
+    /**
+     * Creates a new options array.
+     * 
+     * @param options
+     */
+    @SafeVarargs
+	public OptionUtils(T... options) {
+        this.options = options;
+    }
+
+    /**
+     * Check if the selection option is available
+     * 
+     * @param option
+     * @return .
+     */
+    public boolean contains(Class<? extends T> option) {
+        for (T t : this.options) {
+            if (option.isAssignableFrom(t.getClass())) return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Check if the selection option is available
+     * 
+     * @param option
+     * @return .
+     */
+	@SuppressWarnings("unchecked")
+    public boolean containsAny( Class<? extends T>... option) {
+        for (T t : this.options) {
+            for (Class<? extends T> cls : option) {
+                if (cls.isAssignableFrom(t.getClass())) return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Returns the selection option
+     * 
+     * @param <O> 
+     * 
+     * @param option
+     * @param deflt 
+     * @return .
+     */
+    @SuppressWarnings("unchecked")
+    public <O extends T> O get(Class<? extends O> option, O... deflt) {
+        for (T t : this.options) {
+            if (option.isAssignableFrom(t.getClass())) return (O) t;
+        }
+
+        // return the default if there is any
+        if (deflt.length > 0) return deflt[0];
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;deflt&#39; (index: 1 | type: O[). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;option&#39;. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).